diesel/src/sqlite/connection/stmt.rs

```diff
@@ -25,6 +25,12 @@ impl Statement {
         sql: &str,
         is_cached: PrepareForCache,
     ) -> QueryResult<Self> {
+        if sql.is_empty() {
+            return Err(crate::result::Error::QueryBuilderError(Box::new(
+                crate::result::EmptyQuery,
+            )));
+        }
+
```

Comment:
Should we add a check after the call to `sqlite3_prepare_v3()` that `stmt` is not null before calling `NonNull::new_unchecked()` ?

Reply:
Given the documentation of `sqlite3_prepare_v3` that shouldn't be necessary as we already cover both variants that are documented to return null. That written: It might still be worth to add that additional check to be sure.

Reply:
The above check is not enough, since the pointer will be null if the query string contains only whitespaces or comments. We should also add a test for this.